src/avro_util.erl

```diff
@@ -295,6 +296,113 @@ resolve_duplicated_refs(Type0) ->
   {Type, _Refs} = resolve_duplicated_refs(Type0, []),
   Type.
 
+%% @doc Check if writer schema is compatible to reader schema.
+-spec is_compatible(avro_type(), avro_type()) -> true | {not_compatible, _, _}.
```

Comment:
return spec lies here. function may return `not_compatible | reader_missing_default_value`.
@zmstone I wonder why dialyzer didn't notice that.
 I also think it might make sense to return errors in a form of `{error, Reason} when Reason :: {not_compatible, A, B} | {reader_missing..., A}`, because it makes it a bit easier to match error in `case`.